base/atomics.jl

```diff
@@ -23,7 +23,7 @@ else
                       UInt8, UInt16, UInt32, UInt64, UInt128)
 end
 const floattypes = (Float16, Float32, Float64)
-const arithmetictypes = (inttypes..., floattypes...)
+const arithmetictypes = (inttypes..., floattypes..., Ptr)
 # TODO: Support Ptr
 const atomictypes = (arithmetictypes..., Bool)
```

Comment:
At first, I thought `Ptr` can't be added to `arithmetictypes` for the same reason that `Bool` couldn't: it doesn't support the same sets of operations. (You can't add two `Ptr`s together, etc). But in fact, I think it's fine since `arithmetictypes` is simply controlling _whether_ we'll define `atomic_add!` and `atomic_sub!`, which we do want to do. 👍 So we'll just need a separate `if` case there to define the `Ptr`-specific methods (See my comment below 🙂).

Reply:
I actually removed it from `arithmetictypes` because even though `atomic_add!/sub!` exist for `Ptr`, they're different enough that it makes sense to me to not include it.